fuel-types/src/array_types.rs

```diff
@@ -80,16 +88,30 @@ macro_rules! key_with_big_array {
     };
 }
 
-macro_rules! key_methods {
-    ($i:ident, $s:expr) => {
+macro_rules! key_new_method {
+    ($i:ident, $s:expr,true) => {
         impl $i {
-            /// Memory length of the type
-            pub const LEN: usize = $s;
-
             /// Bytes constructor.
             pub const fn new(bytes: [u8; $s]) -> Self {
                 Self(bytes)
             }
+        }
+    };
+    ($i:ident, $s:expr,false) => {
+        impl $i {
+            /// Bytes constructor
+            pub const fn new_from_bytes(bytes: [u8; $s]) -> Self {
```

Comment:
It is better to have only one the same interface/constructor for all types. We can use `new_from_bytes` for each type because it has the same meaning for all of them. But is is a huge breaking change. Instead, we can continue to use `new` but describe that it creates the type from the bytes without any business logic.